Lib/asyncio/streams.py

```diff
@@ -766,18 +425,17 @@ def get_extra_info(self, name, default=None):
             if exc is not None:
                 raise exc
         if self._transport.is_closing():
-            # Yield to the event loop so connection_lost() may be
-            # called.  Without this, _drain_helper() would return
-            # immediately, and code that calls
-            #     write(...); await drain()
-            # in a loop would never call connection_lost(), so it
-            # would not see an error when the socket is closed.
-            await tasks.sleep(0, loop=self._loop)
```

Comment:
This was added in commit https://github.com/python/cpython/commit/23b4b697e5b6cc897696f9c0288c187d2d24bff2, but it looks like it's unrelated to the API changes. This seems like it would still be quite useful.